### DISCUSSION TITLE
Batch Mode variable made available to interface extensions

### DISCUSSION THREAD

**Discussion by timio23** [directus/directus#17553] [Feature Requests]
### Summary

It would be really useful to have the batchMode & batchActive variables available to the interface extensions so we can alter the interface appearance or options when batch editing records.

I can see they are available in form-field-interface.vue but not passed to the component.

### Basic Example

<pre><code>&lt;component
	:is="..."
	v-if="interfaceExists"
	v-bind="(field.meta && field.meta.options) || {}"
	:autofocus="disabled !== true && autofocus"
	:disabled="disabled"
	:loading="loading"
	:value="modelValue === undefined ? field.schema?.default_value : modelValue"
	:width="(field.meta && field.meta.width) || 'full'"
	:type="field.type"

        :batchMode="batchMode" // Add this
        :batchActive="batchActive" // Add this

	:collection="field.collection"
	:field="field.field"
	:field-data="field"
	:primary-key="primaryKey"
	:length="field.schema && field.schema.max_length"
	@input="$emit('update:modelValue', $event)"
	@set-field-value="$emit('setFieldValue', $event)"
/&gt;</code></pre>

Then in the custom interface.vue

<pre><code>props: {
    batchMode: {
        type: boolean,
        default: false,
    },
    batchActive: {
        type: boolean,
        default: false,
    },
    ....
</code></pre>

### Motivation

The ability to alter the interface appearance or options when batch editing records in order to limit what can be batched.

### Detailed Design

N/A

### Requirements List

N/A

### Drawbacks

No drawback as it's an optional parameter to use within a custom interface

### Alternatives

Tried using conditional options based on field values but these do not work in a batch interface because there is no values to check.

### Adoption Strategy

N/A

**Comment by timio23** [comment]
Made the change to my own installation and can confirm this change will work as expected.

**Comment by rijkvanzanten** [comment]
This was implemented in https://github.com/directus/directus/pull/18624
